sensor/common/registry/registry_store.go

```diff
@@ -122,12 +153,53 @@ func (rs *Store) GetRegistryForImage(image *storage.ImageName) (registryTypes.Re
 		}
 	}
 
-	return nil, errors.Errorf("Unknown image registry: %q", reg)
+	return nil, errors.Errorf("unknown image registry: %q", reg)
 }
 
-// HasRegistryForImage returns true when the registry store has the registry
-// for the given image.
-func (rs *Store) HasRegistryForImage(image *storage.ImageName) bool {
-	reg, err := rs.GetRegistryForImage(image)
-	return reg != nil && err == nil
+// getOrInitGlobalRegistries will return global registries if existing, or create a new set
+// mimics `getRegistries`
+func (rs *Store) getOrInitGlobalRegistries() registries.Set {
```

Comment:
I think https://pkg.go.dev/sync#Once.Do may be good here

Reply:
I was mirroring the existing logic from the `getRegistries` method in the same module - it's also protected by a lock - is there a reason why once.Do is preferred? Assuming we'd want to change the existing logic as well if so.

Reply:
This is a little bit different. `getRegistries` adds new entries to a map while `getOrInitGlobalRegistries` creates a new set once, lazily. I think using `sync.Once` makes sense here. I (believe) it is more efficient than unconditionally locking/unlocking, and works for this use-case. I don't think it would work for the `getRegistries` use-case because we need to create a new registry set for each key, instead of just once, total.

Another option is to just initialize the "global" registries set in the constructor instead of lazily to remove the need for using any kind of synchronization primitives for this structure.

Also, I just realized this is using the same mutex as used for the "namespaced" registries. I think they should use different mutexes since they are unrelated data structures

Reply:
sounds good, i'll make the change to once and separate the locks